optimism-children-indexer/optimism-children-indexer-logic/src/services/layer3_indexer/indexer.rs

```diff
@@ -0,0 +1,278 @@
+//! Layer3 Indexer.
+//!
+//! This module manages concurrent indexing tasks for multiple Layer3 chains.
+//! It automatically spawns, monitors, and restarts indexing tasks based on the current
+//! state of chains in the database.
+use super::{
+    types::{optimism_children_l3_deposits, Layer3IndexerTaskOutput, Layer3IndexerTaskOutputItem},
+    Layer3IndexerTask,
+};
+
+use anyhow::Result;
+use chrono::Utc;
+use optimism_children_indexer_entity::optimism_children_l3_chains::{self, Model as Layer3Chain};
+use sea_orm::{ActiveModelTrait, DatabaseConnection, EntityTrait, Set};
+use std::{collections::HashMap, sync::Arc};
+use tokio::{
+    task::{AbortHandle, JoinSet},
+    time::{interval, Duration},
+};
+
+///  TODO: Consider making this configurable via Settings
+/// Time between database chains refreshes
+const REFRESH_CHAINS_INTERVAL: Duration = Duration::from_secs(15);
+/// Restart delay for fully synced chains
+const RESTART_DELAY_SYNCED: Duration = Duration::from_mins(5);
+/// Restart delay for chains falling behind
+const RESTART_DELAY_BEHIND: Duration = Duration::from_secs(5);
+/// Restart delay for failing chains
+const RESTART_DELAY_FAILING: Duration = Duration::from_secs(90);
+
+/// Main task for Layer3 Indexer.
+pub struct Layer3Indexer {
+    db: Arc<DatabaseConnection>,
+    chains: HashMap<i64, Layer3Chain>,
```

Comment:
`i64` means the chain id, right? And it's `i64` because that's what Alloy uses for chain id?
Can we make it an alias `type ChainId = i64` and use it in our interfaces and types, to make it clear what's expected?